library/Icingadb/Common/LoadMore.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Common;
+
+use Icinga\Module\Icingadb\Widget\ItemList\PageSeparatorItem;
+use Icinga\Module\Icingadb\Widget\ShowMore;
+use ipl\Orm\ResultSet;
+use ipl\Web\Url;
+
+trait LoadMore
+{
+    /** @var int */
+    protected $pageSize;
+
+    /** @var int */
+    protected $pageNumber;
+
+    /** @var Url */
+    protected $loadMoreUrl;
+
+    /**
+     * Set the page size
+     *
+     * @param int $size
+     *
+     * @return $this
+     */
+    public function setPageSize($size)
+    {
+        $this->pageSize = $size;
+
+        return $this;
+    }
+
+    /**
+     * Set the page number
+     *
+     * @param int $number
+     *
+     * @return $this
+     */
+    public function setPageNumber($number)
+    {
+        $this->pageNumber = $number;
+
+        return $this;
+    }
+
+    /**
+     * Set the url for LoadMore widget
+     *
+     * @param Url $url
+     *
+     * @return $this
+     */
+    public function setLoadMoreUrl(Url $url)
+    {
+        $this->loadMoreUrl = $url;
+
+        return $this;
+    }
+
+    /**
+     * Add the page separator and the "LoadMore" button at the desired position
```

Comment:
That's the description of what it additionally does (if it begins with *Adds*), not the main functionality. Keep that with *Adds* as extended description and add a main description what the method returns. And on that note, also add a `@returns Generator` annotation.